components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -361,6 +364,13 @@ public OAuth2AccessTokenRespDTO issue(OAuth2AccessTokenReqDTO tokenReqDTO)
         List<String> requestedAllowedScopes = new ArrayList<>();
         String[] requestedScopes = tokReqMsgCtx.getScope();
         List<String> scopesToBeValidated = new ArrayList<>();
+        String[] requestedOIDCScopes = new String[0];
+        try {
+            //remove OIDC scopes before validation
+            requestedOIDCScopes = OAuth2Util.getRequestedOIDCScopes(requestedScopes);
+        } catch (IdentityOAuthAdminException e) {
+            log.error("Unable to retrieve OIDC Scopes." + e.getMessage());
```

Comment:
If this error happens can we continue?

If not we have to thrown an exception and break the flow

Reply:
if we cant filter oidc scope, we can not continue. can we throw `IdentityException` here?

Reply:
Check for a type already thrown and use it